src/NetAppFiles/NetAppFiles/Snapshot/RestoreNetAppFilesSnapshotFiles.cs

```diff
@@ -0,0 +1,157 @@
+            nameof(PoolName))]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "List of files to be restored")]
+        [ValidateNotNullOrEmpty]
+        public string[] FilePath { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Destination folder where the files will be restored")]
+        [ValidateNotNullOrEmpty]
+        public string DestinationPath { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ResourceIdParameterSet,
+            HelpMessage = "The resource id of the ANF snapshot")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+        
+        [Parameter(
+            ParameterSetName = ParentObjectParameterSet,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = "The volume object containing the snapshot")]
+        [ValidateNotNullOrEmpty]
+        public PSNetAppFilesVolume VolumeObject { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Return whether the specified files where successfully restored")]
+        public SwitchParameter PassThru { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            bool success = false;
+            if (ParameterSetName == ResourceIdParameterSet)
+            {
+                var resourceIdentifier = new ResourceIdentifier(ResourceId);
+                ResourceGroupName = resourceIdentifier.ResourceGroupName;
+                var parentResources = resourceIdentifier.ParentResource.Split('/');
+                AccountName = parentResources[1];
+                PoolName = parentResources[3];
+                VolumeName = parentResources[5];
+                Name = resourceIdentifier.ResourceName;
+            }
+            else if (ParameterSetName == ParentObjectParameterSet)
+            {
+                ResourceGroupName = VolumeObject.ResourceGroupName;
+                var NameParts = VolumeObject.Name.Split('/');
+                AccountName = NameParts[0];
+                PoolName = NameParts[1];
+                VolumeName = NameParts[2];
+            }
+
+            if (ShouldProcess(Name, string.Format(PowerShell.Cmdlets.NetAppFiles.Properties.Resources.RestoringFilesMessage, string.Join(",",FilePath))))
+            {
+                SnapshotRestoreFiles snapshotRestoreFilesBody = new SnapshotRestoreFiles()
+                {
+                    FilePaths = FilePath,
+                    DestinationPath = DestinationPath
+                };
+
+                AzureNetAppFilesManagementClient.Snapshots.RestoreFiles(snapshotRestoreFilesBody, ResourceGroupName, AccountName, PoolName, VolumeName, Name);
+                success = true;
+            }
+            if (PassThru)
```

Comment:
```suggestion
            if (PassThru.IsPresent)
```
Not check the syntax whether successes